Plugin crash/panic after update of "monitor_arn_list" argument in "aws_ce_anomaly_subscription" resource

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform v1.2.5
on linux_amd64
provider registry.terraform.io/hashicorp/aws v4.23.0

### Affected Resource(s)

* aws_ce_anomaly_subscription

### Terraform Configuration Files

```hcl
resource "aws_ce_anomaly_monitor" "Monitor1" {
  name                  = "Monitor1"
  monitor_type          = "DIMENSIONAL"
  monitor_dimension     = "SERVICE"
}

resource "aws_ce_anomaly_monitor" "Monitor2" {
  name                  = "Monitor2"
  monitor_type          = "CUSTOM"
	monitor_specification = <<JSON
{
	"And": null,
	"CostCategories": null,
	"Dimensions": {
		"Key": "LINKED_ACCOUNT",
		"Values": ["222222222222"],
		"MatchOptions": ["EQUALS"]
	},
	"Not": null,
	"Or": null,
	"Tags": null
}
JSON
}

resource "aws_ce_anomaly_monitor" "Monitor3" {
  name                  = "Monitor3"
  monitor_type          = "CUSTOM"
	monitor_specification = <<JSON
{
	"And": null,
	"CostCategories": null,
	"Dimensions": {
		"Key": "LINKED_ACCOUNT",
		"Values": ["333333333333"],
		"MatchOptions": ["EQUALS"]
	},
	"Not": null,
	"Or": null,
	"Tags": null
}
JSON
}

resource "aws_ce_anomaly_subscription" "Subscription1" {
  name             = "Subscription1"
  frequency        = "DAILY"

  monitor_arn_list = [
    aws_ce_anomaly_monitor.Monitor1.arn,
    aws_ce_anomaly_monitor.Monitor2.arn
	]

  subscriber {
    type    = "EMAIL"
    address = "foo@bar.com"
  }

  threshold  = 5
}
```

### Debug Output
Will provide debug output only if panic output is not sufficient.

### Panic Output
```
Stack trace from the terraform-provider-aws_v4.23.0_x5 plugin:

panic: interface conversion: interface {} is []interface {}, not []string

goroutine 38 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ce.resourceAnomalySubscriptionUpdate({0xab8eaa8, 0xc0030cd800}, 0xc000635f80, {0x84c7d60, 0xc000833500})
	github.com/hashicorp/terraform-provider-aws/internal/service/ce/anomaly_subscription.go:187 +0xf5e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000bf2460, {0xab8eae0, 0xc00234dce0}, 0xd, {0x84c7d60, 0xc000833500})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.19.0/helper/schema/resource.go:741 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000bf2460, {0xab8eae0, 0xc00234dce0}, 0xc0023ea340, 0xc000635600, {0x84c7d60, 0xc000833500})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.19.0/helper/schema/resource.go:847 +0x9e5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002161b0, {0xab8eae0, 0xc00234db60}, 0xc0030de4b0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.19.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc001baa180, 0xc001baa1e0, {0xc0027d4f40, 0x2, 0x2}, 0xc001baa1b0, 0xc001bb8ab0, 0xc0020c6650, 0xc001baa210}, {0xab8eae0, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.7.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x142
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0014cc780, {0xab8eae0, 0xc00234c5d0}, 0xc0030276c0)
	github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov5/tf5server/server.go:813 +0x551
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x971ff00, 0xc0014cc780}, {0xab8eae0, 0xc00234c5d0}, 0xc003027650, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0021fd880, {0xaca0a30, 0xc002415380}, 0xc001ddc6c0, 0xc0028427b0, 0x10b7e3c0, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1295 +0xb03
google.golang.org/grpc.(*Server).handleStream(0xc0021fd880, {0xaca0a30, 0xc002415380}, 0xc001ddc6c0, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1636 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.48.0/server.go:930 +0x294

Error: The terraform-provider-aws_v4.23.0_x5 plugin crashed!
```

### Expected Behavior
`monitor_arn_list` for the `aws_ce_anomaly_subscription` resource be updated successfully with changes to that list.


### Actual Behavior
panic error on apply

### Steps to Reproduce

1. `terraform apply`, should be successful
2. Edit `monitor_arn_list` argument for the `aws_ce_anomaly_subscription` resource, either by adding to or removing monitors from the list.
3. `terraform apply`, modification/update of `aws_ce_anomaly_subscription` required, should fail.

### Important Factoids
Cost Explorer must be "activated". Go into the Cost Explorer console, wait 24hrs+.

### References
Looks very similar to this issue: https://github.com/hashicorp/terraform-provider-aws/issues/24881

--- a/.changelog/25941.txt
+++ b/.changelog/25941.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_ce_anomaly_subscription:Fix crash upon adding or removing monitor ARNs to `monitor_arn_list`.
+```

--- a/internal/service/ce/anomaly_subscription.go
+++ b/internal/service/ce/anomaly_subscription.go
@@ -184,7 +184,7 @@ func resourceAnomalySubscriptionUpdate(ctx context.Context, d *schema.ResourceDa
 	}
 
 	if d.HasChange("monitor_arn_list") {
-		input.MonitorArnList = aws.StringSlice(d.Get("monitor_arn_list").([]string))
+		input.MonitorArnList = aws.StringSlice(expandAnomalySubscriptionMonitorARNList(d.Get("monitor_arn_list").([]interface{})))
 		requestUpdate = true
 	}
 

--- a/internal/service/ce/anomaly_subscription_test.go
+++ b/internal/service/ce/anomaly_subscription_test.go
@@ -432,7 +432,7 @@ resource "aws_ce_anomaly_monitor" "test2" {
 resource "aws_ce_anomaly_subscription" "test" {
   name      = %[1]q
   threshold = 100
-  frequency = "DAILY"
+  frequency = "WEEKLY"
 
   monitor_arn_list = [
     aws_ce_anomaly_monitor.test.arn,

--- a/internal/service/ce/anomaly_subscription_test.go
+++ b/internal/service/ce/anomaly_subscription_test.go
@@ -120,7 +120,6 @@ func TestAccCEAnomalySubscription_MonitorARNList(t *testing.T) {
 	resourceName := "aws_ce_anomaly_subscription.test"
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	rName2 := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
-	rName3 := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	domain := acctest.RandomDomainName()
 	address := acctest.RandomEmailAddress(domain)
 
@@ -143,7 +142,7 @@ func TestAccCEAnomalySubscription_MonitorARNList(t *testing.T) {
 				ImportStateVerify: true,
 			},
 			{
-				Config: testAccAnomalySubscriptionConfig_monitorARNList(rName2, rName3, address),
+				Config: testAccAnomalySubscriptionConfig_monitorARNList(rName, rName2, address),
 				Check: resource.ComposeAggregateTestCheckFunc(
 					testAccCheckAnomalySubscriptionExists(resourceName, &subscription),
 					resource.TestCheckResourceAttrPair(resourceName, "monitor_arn_list.0", "aws_ce_anomaly_monitor.test", "arn"),
@@ -433,7 +432,7 @@ resource "aws_ce_anomaly_monitor" "test2" {
 resource "aws_ce_anomaly_subscription" "test" {
   name      = %[1]q
   threshold = 100
-  frequency = "WEEKLY"
+  frequency = "DAILY"
 
   monitor_arn_list = [
     aws_ce_anomaly_monitor.test.arn,

